File: Others/c, c++/insertion_sort.cpp
#include<iostream>
using namespace std;

void insertionSort(int arr[], int n){
    int i, key, j;
        for (i = 1; i < n; i++){
            key = arr[i];
            j = i-1;
            while (j >= 0 && arr[j] > key){
                arr[j+1] = arr[j];
                j = j-1;
            }
            arr[j+1] = key;
        }
}

int main(){

    int arr[5]={0,8,-1,4,3};
    insertionSort(arr,5);
    for(int i=0 ; i<5; i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}

File: DSA/sort/insertionSort.js
const insertionSort = (arr) => {
    for (let i = 1; i < arr.length; i++) {
        let tmp = arr[i]
        let j = i - 1;
        while (j >= 0 && tmp < arr[j]) {
            arr[j + 1] = arr[j]
            j--;
        }
        arr[j + 1] = tmp;
    }
    return arr
}
console.log(insertionSort([2, 3, 1]));